modules/katello_repository_set.py

```diff
@@ -0,0 +1,188 @@
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+
+def get_desired_repos(desired_substitutions, available_repos):
+    desired_repos = []
+    for sub in desired_substitutions:
+        desired_repos += filter(lambda available: available['substitutions'] == sub, available_repos)
+    return desired_repos
+
+
+def repository_set(module, name, organization, product, state, repositories=[]):
+    changed = False
+    organization = find_organization(module, organization)
+    product = find_product(module, product, organization)
+    repo_set = find_repository_set(module, name=name, product=product)
+
+    available_repos = repo_set.available_repositories()['results']
+    current_repos = map(lambda r: r.read(), repo_set.repositories)
+    desired_repos = get_desired_repos(repositories, available_repos)
+
+    available_repo_names = set(map(lambda r: r['repo_name'], available_repos))
+    current_repo_names = set(map(lambda r: r.name, current_repos))
+    desired_repo_names = set(map(lambda r: r['repo_name'], desired_repos))
+
+    if len(desired_repo_names - available_repo_names) > 0:
+        module.fail_json(msg="Desired repositories are not available on the repository set {}. Desired: {} Available: {}"
+                         .format(name, desired_repo_names, available_repo_names))
+
+    if state == 'enabled':
+        for repo in desired_repo_names - current_repo_names:
+            repo_to_enable = (r for r in available_repos if r['repo_name'] == repo).next()
+            repo_set.enable(data=repo_to_enable['substitutions'])
+            changed = True
+    else:
+        for repo in current_repo_names & desired_repo_names:
+            repo_to_disable = (r for r in available_repos if r['repo_name'] == repo).next()
+            repo_set.disable(data=repo_to_disable['substitutions'])
+            changed = True
+    return changed
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True, no_log=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(type='bool', default=False),
+            name=dict(required=True),
+            product=dict(),
+            organization=dict(required=True),
+            repositories=dict(type='list'),
+            state=dict(required=True),
+        )
+    )
+
+    if not HAS_NAILGUN_PACKAGE:
+        module.fail_json(msg="Missing required nailgun module (check docs or install "
+                             "with: pip install nailgun)")
+
+    server_url = module.params['server_url']
+    username = module.params['username']
+    password = module.params['password']
+    verify_ssl = module.params['verify_ssl']
+    name = module.params['name']
+    product = module.params['product']
+    organization = module.params['organization']
+    repositories = module.params['repositories'] if module.params['repositories'] is not None else []
```

Comment:
If I don't specify any repositories, what can this module do? That is, I feel like this is the key param to the whole module and might should be required.